File: simple_cases/sediment_rip/postprocessing/plot_eta_c_dep.m
% Clearing Matlab workspace
clear all

% -----------------------
% ----- User Input ------
% -----------------------

fdir='../work/output/'


% -----------------------
% -- End of user input --
% -----------------------

% Getting depth file and determining domain dimensions

dep=load([fdir 'dep.out']);
[n,m]=size(dep);
dx = 1;
dy = 2;
x=[0:m-1]*dx;
y=[0:n-1]*dy;

% Generating 2D grid with x and y points
[xx,yy]=meshgrid(x,y);


% Plot window dimensions
set(gcf,'units','inches','paperunits','inches','papersize', [7 17],'position',[1 1 9 4],'paperposition',[0 0 9 4]);


%nstart=input('nstart');
%nend=input('nend');

nstart=5;
nend=5;
% previous version (2nd revision) nstart=280

icount=0;
for num=nstart:nend
    
% Padding integer values with zeros
% to be 5 letters long e.g. 1 -> 00001
icount=icount+1;
fnum=sprintf('%.5d',num);
% Loading data from files

u=load([fdir 'umean_' fnum]);
v=load([fdir 'vmean_' fnum]);
eta=load([fdir 'eta_' fnum]);
mask=load([fdir 'mask_' fnum]);
ch=load([fdir 'C_' fnum]);
dep1=load([fdir 'dep_' fnum]);
depp=dep1-dep;

% Removing masked regions from plot
eta(mask<1)=NaN;
dep(mask<1)=NaN;
u(mask<1)=NaN;
v(mask<1)=NaN;

ax=[0 249 0 199];
clf

subplot(131)
% plot eta
hp=pcolor(xx,yy,eta);shading interp
caxis([-0.6 1.2])
colormap(jet)
h_bar=colorbar('location','SouthOutside');
set(get(h_bar,'xlabel'),'string','\eta (m)' )

xlabel('x (m)')
ylabel('y (m)')

%axis image, 
axis(ax)

% -------------------
subplot(132)

pcolor(xx,yy,log10(ch)),shading interp;
caxis([-1 1])
h_bar=colorbar('location','SouthOutside');
set(get(h_bar,'xlabel'),'string','log10(C) (g/l)' )

xlabel('x (m)')
%ylabel('y (m)')
hold on

s=10;
sx=4;
sy=4;
quiver(xx(1:sy:end,1:sx:end),yy(1:sy:end,1:sx:end),s*u(1:sy:end,1:sx:end),s*v(1:sy:end,1:sx:end),0,'k')


%axis image, 
axis(ax)

subplot(133)

contourf(xx,yy,depp,10);
%caxis([-12 0])
h_bar=colorbar('location','SouthOutside');
set(get(h_bar,'xlabel'),'string','depth change (m)' )
%axis image, 
axis(ax)
caxis([-0.05 0.05])

xlabel('x (m)')
%ylabel('y (m)')

pause(0.1)
end

File: simple_cases/sediment_rip/postprocessing/plot_eta_c_dep.py

# -----------------------
# ----- User Input ------
# -----------------------

import os
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
from matplotlib.cm import jet
from matplotlib import ticker

HOME = os.environ['HOME']
fdir = os.path.join(HOME,'FUNWAVE-TVD','simple_cases','sediment_rip','work','output')

# -----------------------
# -- End of user input --
# -----------------------

# Getting depth file and determining domain dimensions

depFile = os.path.join(fdir,'dep.out')
dep = np.loadtxt(depFile)
[n,m] = dep.shape

# discretization
dx = 1.0
dy = 2.0

# x and y field vectors 
x = np.arange(0,m*dx,dx)
y = np.arange(0,n*dy,dy)

# Generating 2D grid with x and y points
[xx,yy] = np.meshgrid(x,y)

nstart = input('enter nstart: ')
nend = input('enter nend: ')

icount = 0
for num in range(int(nstart),int(nend)+1):
              # Padding integer values with zeros
              # to be 5 letters long e.g. 1 -> 00001
              icount = icount + 1
              fnum = '{0:0>5}'.format(num)

              # Loading data from files
              u = np.loadtxt(os.path.join(fdir,'umean_'+fnum))
              v = np.loadtxt(os.path.join(fdir,'vmean_'+fnum))
              eta = np.loadtxt(os.path.join(fdir,'eta_'+fnum))
              mask = np.loadtxt(os.path.join(fdir,'mask_'+fnum))
              ch = np.loadtxt(os.path.join(fdir,'C_'+fnum))
              dep1 = np.loadtxt(os.path.join(fdir,'dep_'+fnum))
              depp = dep1-dep

              # Removing masked regions from plot
              eta[np.where( mask < 1)] = np.nan
              dep[np.where( mask < 1)] = np.nan
              u[np.where( mask < 1)] = np.nan
              v[np.where( mask < 1)] = np.nan

              # ------------------
              plot1 = plt.subplot(1,3,1)
              plot1.tick_params(axis='both', which='major', labelsize=6)
              plot1.tick_params(axis='both', which='minor', labelsize=6)

              # plot eta (surface elevation)
              hp = plt.pcolor(xx,yy,eta, cmap=jet)
              plot1.axis([0,7*m*dx/9,0,(n-1)*dy])
              h_bar = plt.colorbar(hp,orientation="horizontal")
              h_bar.ax.tick_params(labelsize=6)
              plt.clim(-0.5, 1.0)
              tick_locator = ticker.MaxNLocator(nbins=5)
              h_bar.locator = tick_locator
              h_bar.update_ticks()

              h_bar.ax.set_xlabel(r'$\eta (m)$')
              plt.xlabel('x (m)')
              plt.ylabel('y (m)')

              # ----------------
              plot2 = plt.subplot(1,3,2)
              plot2.tick_params(axis='both', which='major', labelsize=6)
              plot2.tick_params(axis='both', which='minor', labelsize=6)

              cf = plt.pcolor(xx,yy,np.log10(ch));

              c_bar = plt.colorbar(cf, orientation='horizontal')
              c_bar.ax.tick_params(labelsize=6)
              plt.clim(-1.0,1.0)
              tick_locator = ticker.MaxNLocator(nbins=5)
              c_bar.locator = tick_locator
              c_bar.update_ticks()

              c_bar.ax.set_xlabel('log10(C) (g/l)')

              everyWhich=4 # plot every nth arrow (scales with image)
              q = plt.quiver(xx[::everyWhich, ::everyWhich],
                             yy[::everyWhich, ::everyWhich],
                             u[::everyWhich, ::everyWhich],
                             v[::everyWhich, ::everyWhich])

              plot2.axis([0,7*m*dx/9,0,(n-1)*dy])
              plt.xlabel('x (m)')


              # ---------------
              plot3 = plt.subplot(1,3,3)
              plot3.tick_params(axis='both', which='major', labelsize=6)
              plot3.tick_params(axis='both', which='minor', labelsize=6)

              db = plt.contourf(xx,yy,depp,10)
              few_contours = plt.contour(xx,yy,depp,10)
              plot3.axis([0,7*m*dx/9,0,(n-1)*dy])
              d_bar = plt.colorbar(db, orientation="horizontal")
              d_bar.ax.tick_params(labelsize=6)
              plt.clim(-0.05, 0.05)
              tick_locator = ticker.MaxNLocator(nbins=5)
              d_bar.locator = tick_locator
              d_bar.update_ticks()

              d_bar.ax.set_xlabel('depth change (m)')
              plt.xlabel('x (m)')
              
              plt.savefig('sediment_rip_num_'+str(num)+'.png', dpi=400)                                                            
              plt.show()                                                                                                              
